pages/api/me/password.js

```diff
@@ -24,27 +24,33 @@ SOFTWARE.
 
 import { withSession } from '@klaudsol/commons/lib/Session';
 import { defaultErrorHandler } from '@klaudsol/commons/lib/ErrorHandler';
-import { assert } from '@klaudsol/commons/lib/Permissions';
 import { handleRequests } from '@klaudsol/commons/lib/API';
 import { OK, UNPROCESSABLE_ENTITY } from '@klaudsol/commons/lib/HttpStatuses';
 import Session from '@klaudsol/commons/models/Session';
 import People from '@klaudsol/commons/models/People';
 import RecordNotFound from '@klaudsol/commons/errors/RecordNotFound';
+import UnauthorizedError from '@klaudsol/commons/errors/UnauthorizedError';
+import { verifyToken } from '@klaudsol/commons/lib/JWT';
+import { getCookie } from 'cookies-next';
 
 export default withSession(handleRequests({ put }));
 
 async function put(req, res) { 
  try{
+    const cookie = getCookie('token', { req, res });
+    if (!cookie) throw new UnauthorizedError();
 
-    await assert({
-     loggedIn: true,
-    }, req);
+    const { sessionToken } = verifyToken(cookie);
```

Comment:
for JWT tokens, instead of putting them in a cookie, I suggest just outputing it as a JSON reply. This way, any clients, even those not supporting cookies (e.g. mobile phones) can use the endpoint. 

It's up to the client to know where to store it, and to pass it onto the next request.